src/SignalR/server/StackExchangeRedis/test/TestConnectionMultiplexer.cs

```diff
@@ -364,7 +377,14 @@ public void WaitAll(params Task[] tasks)
 
         public ChannelMessageQueue Subscribe(RedisChannel channel, CommandFlags flags = CommandFlags.None)
         {
-            throw new NotImplementedException();
+            var redisSubscriberType = typeof(RedisChannel).Assembly.GetType("StackExchange.Redis.RedisSubscriber");
+            var ctor = typeof(ChannelMessageQueue).GetConstructor(BindingFlags.Instance | BindingFlags.NonPublic,
+                binder: null,
+                new Type[] { typeof(RedisChannel).MakeByRefType(), redisSubscriberType }, modifiers: null);
+
+            var queue = (ChannelMessageQueue)ctor.Invoke(new object[] { channel, null });
```

Comment:
I feel like this requires far more explanation. This looks super fragile.

Reply:
The `ChannelMessageQueue` type can't be mocked and has a private ctor, so we need to private reflect to get unit tests working.

Reply:
Why didn't we need to do this before?

Reply:
Because this type didn't use to exist. The tests weren't skipped back in version 1.x of stackexchange.redis, but with the move to 2.x we had to skip them because of this type.

Reply:
Is there an issue in stackexchange.redis to make `ISubscriber` more mockable? In the meantime, I think we should create our own `ISignalRSubscriber` and `ISignalRChannelMessageQueue` interfaces and use those.

Reply:
Looking at it more, it might not be worth it. Let's make sure an issue is filed though.

Reply:
https://github.com/StackExchange/StackExchange.Redis/issues/969

Reply:
Oh, so it's definitely going to be fixed then 😆